### ISSUE TITLE
Preview mode state of editors not conveyed to screen reader

### ISSUE THREAD

**Issue by ndarilek** [microsoft/vscode#98180]
Didn't realize that tabs were preview by default until edited, which threw me when I opened a project and attempted to set up common files in a few tabs/groups and my editors kept getting replaced.

To make this behavior a bit less surprising, I'd suggest appending "Preview" to any preview-mode tabs. I'd also suggest speaking a message when the tab transitions to edit mode to make the change more discoverable. It only happens once AFAIK, and is important enough that it should be conveyed to the user.

Thanks.

**Comment by isidorn** [comment]
@ndarilek makes perfect sense to improve this as you suggest
@bpasero can we add `preview` to the tab `aria-label` when the editor is in preview?

**Comment by bpasero** [comment]
@isidorn not clear to me where to add this, only for tabs? Then it would go here?

https://github.com/microsoft/vscode/blob/452dc54c780481129c0d57ac5c0908c0a573a622/src/vs/workbench/browser/parts/editor/tabsTitleControl.ts#L891

More thoughts:
* we recently introduced "pinned" tabs, so it probably needs to be announced too

**Comment by isidorn** [comment]
@bpasero if I remember correctly we try to set the same ariaLabel on the tab and the editor so screen readers smartly de-dup the message read once the focus is set on the editor.
Thus I suggest that we add both "pinned" and "preview" both on the tab aria label and on the editor aria label. Implementation wise both those places should get the aria label from the same place (living on the editor I guess).

**Comment by bpasero** [comment]
Sounds good.

**Comment by isidorn** [comment]
@bpasero do you plan to look into this?
I can also add it. Thanks

**Comment by bpasero** [comment]
I can review a PR but will not have a lot of time to look into it.

**Comment by isidorn** [comment]
@bpasero thanks! I have created https://github.com/microsoft/vscode/pull/98236

**Comment by isidorn** [comment]
This should be fixed now.
@ndarilek you can try it out with VS Code insiders from Thursday.

To verify:
1. Have an editor open in "preview" mode and make sure that the Screen Reader reads "preview" after the editor title once the editor is focused
2. Have an editor tab pinned and make sure that the Screen Reader reads "pinned"  after the editor title once the editor is focused
